dmtools-core/src/main/java/com/github/istin/dmtools/qa/TestCasesGenerator.java

```diff
@@ -118,7 +122,40 @@ public TestCasesResult generateTestCases(TicketContext ticketContext, String ext
 
         List<ITicket> finaResults = findAndLinkSimilarTestCasesBySummary(ticketContext, listOfAllTestCases, true, params.getRelatedTestCasesRules(), params.getTestCaseLinkRelationship());
 
-        List<TestCaseGeneratorAgent.TestCase> newTestCases = testCaseGeneratorAgent.run(new TestCaseGeneratorAgent.Params(params.getTestCasesPriorities(), ToText.Utils.toText(finaResults), ticketContext.toText(), extraRules));
+        // Initialize accumulator for all generated test cases
+        List<TestCaseGeneratorAgent.TestCase> allGeneratedTestCases = new ArrayList<>();
```

Comment:
low: Using a magic number like `2` for division can make the code harder to understand. It's better to define a constant with a descriptive name to clarify the purpose of this division.
```suggestion
        int storyTokens = new Claude35TokenCounter().countTokens(ticketContext.toText());
        int systemTokenLimits = chunkPreparation.getTokenLimit();
        final int CHUNKS_COUNT = 2;
        int tokenLimit = (systemTokenLimits - storyTokens)/CHUNKS_COUNT;
```